### ISSUE TITLE
Unrecoverable error. Failure when converting long string to UTF-16 error_code = 1; retcode = 0; StringData.size = 217; StringData.data = “生”逢盛世，“日”斁AAA in /Users/cm/Realm/realm-java/realm/realm-library/src/main/cpp/io_realm_internal_UncheckedRow.cpp line

### ISSUE THREAD

**Issue by 09ae9c** [realm/realm-java#6563]
```
Unrecoverable error. Failure when converting long string to UTF-16 error_code = 1; retcode = 0; StringData.size = 217; StringData.data = “生”逢盛世，“日”斁AAA in /Users/cm/Realm/realm-java/realm/realm-library/src/main/cpp/io_realm_internal_UncheckedRow.cpp line 161
```

I saw some closed issues have already report the same problem and you said it was fixed, but I still got this problem at version 5.8.0.

https://github.com/realm/realm-java/issues/4343  is this the same problem?

#### Version of Realm and tooling

Realm version(s): 5.8.0

Realm Sync feature enabled: No

Android Studio version: 3.4.1

Android Build Tools version: 28.0.0

Gradle version: 5.1.1

Which Android version and device(s): 9.0

**Comment by cmelchior** [comment]
Possibly. Is this something you can reproduce?

**Comment by 09ae9c** [comment]
> Possibly. Is this something you can reproduce?

I've tried to copy the full text from the error report to my local environment but nothing happened, and I found this crash occurred in query data from Realm(not in write process)


```
at io.realm.internal.UncheckedRow.nativeGetString(Native Method)
	at io.realm.internal.UncheckedRow.getString(UncheckedRow.java:157)
	at io.realm.com_bearyinnovative_horcrux_core_data_personal_model_MessageRealmProxy.realmGet$text(com_bearyinnovative_horcrux_core_data_personal_model_MessageRealmProxy.java:548)
	at com.bearyinnovative.horcrux.core.data.personal.model.Message.getText(Message.kt:56)
```

I have a RealmObject named `Message` and `text` is a String field in `Message`，when this crash occurred, I have already had a `Message` reference, and I just want to access the `text` field.

Additional, this crash only happened in our production version, it means we enabled Encryption in `RealmConfiguration`, is it possible that this crash related to `encryptionKey` field?

**Comment by 09ae9c** [comment]
any update?

**Comment by cmelchior** [comment]
Since this crash happend on an encrypted Realm I'm going to assume it was fixed by our 6.0.1 release: https://github.com/realm/realm-java/blob/master/CHANGELOG.md#6012019-11-11

If this turns out to not be the case, feel free to re-open the issue.
